string ns
string serial_no
bool is_advanced
